### PR TITLE
plotAbundanceDensity: added boxplot and violin

### PR DESCRIPTION
**Pull request by 0xMuluh**
ping: #142

### CODE REVIEW

**Review by antagomir — APPROVED** [review]

**Review by antagomir — APPROVED** [review]

**Comment by antagomir** [comment]
DId you add (at least a short) description of the plotting options in the roxygen documentation? I did not find this.

**Comment by 0xMuluh** [comment]
> DId you add (at least a short) description of the plotting options in the roxygen documentation? I did not find this.

Thanks. I have added to the options.

**Comment by antagomir** [comment]
It is good to update documentation by default always.

**Review by antagomir — APPROVED** [review]

**Comment by antagomir** [comment]
Some checks fail?

**Comment by 0xMuluh** [comment]
> Some checks fail?

The checks are unrelated to this PR. I could prepare another PR with a fix.

**Comment by 0xMuluh** [comment]
I think, this is good to go.

**Review by TuomasBorman — CHANGES_REQUESTED** [review]
Let's rethink this. It seems that the desired plot types are already available in SE ecosystem. We should avoid overlapping work. Check it these plots fulfill the need

```
library(miaViz)

data("peerj13075", package = "mia")
tse <- peerj13075

tse <- transformAssay(tse, method = "relabundance")

plotAbundanceDensity(tse, assay.type = "relabundance", n = 5, layout = "violin")
plotAbundanceDensity(tse, assay.type = "relabundance", n = 5, layout = "box")

library(scater)

top <- getTop(tse)

plotExpression(tse, features = top, assay.type = "relabundance") + ggplot2::coord_flip()
plotExpression(tse, features = top, assay.type = "relabundance", show_violin = FALSE, show_box = TRUE) + ggplot2::coord_flip()


```

**Comment by 0xMuluh** [comment]
> plotExpression(tse, features = top, assay.type = "relabundance", show_violin = FALSE, show_box = TRUE) + ggplot2::coord_flip()

Thanks.

**Comment by 0xMuluh** [comment]
Closing issue now.

**Comment by antagomir** [comment]
Can you summarize what was done regarding the point on plotExpression vs. plotAbundanceDensity, @Daenarys8 ?

**Comment by 0xMuluh** [comment]
> Can you summarize what was done regarding the point on plotExpression vs. plotAbundanceDensity, @Daenarys8 ?

plotExpression  provides the desired layouts `violin` and `boxplot` which can be used to measure counts based on selected taxa.

**Comment by antagomir** [comment]
Can you summarize briefly how this was implemented? Does plotAbundanceDensity internally call plotExpression?

**Comment by 0xMuluh** [comment]
> Can you summarize briefly how this was implemented? Does plotAbundanceDensity internally call plotExpression?

I had a misunderstanding of the intention. I have added the changes.

**Comment by TuomasBorman** [comment]
> Can you summarize briefly how this was implemented? Does plotAbundanceDensity internally call plotExpression?

Does it have to call it internally? It causes more maintaining work comparing to if we just rely on plotExpression?

**Comment by antagomir** [comment]
> > Can you summarize briefly how this was implemented? Does plotAbundanceDensity internally call plotExpression?
> 
> Does it have to call it internally? It causes more maintaining work comparing to if we just rely on plotExpression?

Yes it might be so. 

I did not mean to suggest how it should be done, just asked for a summary on what was done as I could not immediately figure it out from the sources.

Good to discuss first how to proceed, then proceed.

If plotExpression already solves the issue, then the following actions could be useful:

- mention this in the @seealso field of plotAbundanceDensity and perhaps even in its @examples field (for comparison)
- have examples of both in OMA in the same place so that it is easy to find; not all variants have to be shown in OMA, just those we like to recommend by default (+ mention about the other available options)

Did I miss something?

**Comment by 0xMuluh** [comment]
I have mentioned in @seealso and @examples.
Now for OMA, I could add to chapter 10.1, It is the most suitable place since it mentions abundance plots and jitter and density plots are available there.

**Comment by TuomasBorman** [comment]
Good place for plotExpression is also differential abundance chapter as it does not have abundance visualization yet. It is very common to visualize the significant taxa. https://microbiome.github.io/OMA/docs/devel/pages/differential_abundance.html

There could be a link in QC chapter. "Abundances can be visualized also with plotExpression(). See example from here"

**Review by TuomasBorman — APPROVED** [review]
